File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxc=0,maxs=0;
        for(int i =1 ; i< prices.length ; i++)
        {
            maxc+=prices[i]-prices[i-1];
            maxc=Math.max(0,maxc);
            maxs=Math.max(maxc,maxs);
        }
        return maxs;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxp=prices[len(prices)-1]
        profit=0
        for i in prices[::-1]:
            if maxp - i > profit:
                profit= maxp - i
            if i > maxp:
                maxp=i
        return profit